### ISSUE TITLE
bad ../physical_package_id value

### ISSUE THREAD

**Issue by StephaneRenouard** [avafinger/bpi-m64-firmware#5]
When I cat: 
cat  /sys/devices/system/cpu/cpu0/topology/physical_package_id
I got a return value of -1, which is not correct. 

The same for each cpu [0...3]. 

FYI, I'm using kernel 3.10.102

**Comment by avafinger** [comment]
**physical_package_id**: physical package id of cpu#. 
Typically	corresponds to a physical socket number, but the actual value is architecture and platform dependent.

Which values would you expect?
I just typed this in my Intel box and they all return **0**

**Comment by StephaneRenouard** [comment]
As this processor has 4 cores, I should expected 0,1,2,3 or 0,0,0,0 for cpu0, cpu1, cpu2 and cpu3. 
Theses values are used for thread mapping in software. 
I'm trying to run Aerospike software on BPI M64, and this param is checked at runtime. "-1" forbid the software to run. 
I cheated with a fake physical_package_id file, but this could be a drawback in the futur.

**Comment by avafinger** [comment]
Can you please try this and see if still the same? It is kernel 3.10.105.
https://github.com/avafinger/bpi-m64-firmware-v2
PS: try with the SD CARD if you don't want to format your eMMC.

**Comment by avafinger** [comment]
Interesting read:  https://patchwork.kernel.org/patch/9265455/

**Comment by avafinger** [comment]
Kernel 4.11
cat /sys/devices/system/cpu/cpu0/topology/physical_package_id 
0

Will look into it when possible.

**Comment by StephaneRenouard** [comment]
I'll take the time to try 3.10.105 this weekend. Did you try Kernel 4.11 on BPIM64 ?

**Comment by StephaneRenouard** [comment]
Ok, I updated the board from 3.10.102 to 3.10.105. 

Got the same issue.

**Comment by avafinger** [comment]
yeah, 3.10.x seems to get this value as default for ARM on A64. tested also on Armbian on another A64 (Pine64+). Mainline 4.11 (Armbian) was on other A64 board (OPI). Mainline 4.11.x work bare bone on A64, has no HDMI,CSI  and Desktop but you get GbE. 4.11 seems the way to go in this case, but to get fully working, you may need to wait several months....

**Comment by StephaneRenouard** [comment]
I'll take the time to try 3.10.105 this weekend. Did you try Kernel 4.11 on BPIM64 ? 

Update: same issue with 3.10.105. I should have to wait for 4.11...

**Comment by avafinger** [comment]
If you care only about command line and connectivity (eth0), not sure it eMMC works in 4.11 but seems it is working (i still in the learning curve with 4.x), maybe i get it running in a few weeks, not a promise....

**Comment by StephaneRenouard** [comment]
Take your time, i'll try the UP board next week from my side.
